--- a/drivers/net/wireless/bcm4336/dhd_gpio.c
+++ b/drivers/net/wireless/bcm4336/dhd_gpio.c
@@ -1,4 +1,3 @@
-
 #include <osl.h>
 #include <dngl_stats.h>
 #include <dhd.h>
@@ -12,6 +11,10 @@
 #define	sdmmc_channel	s3c_device_hsmmc0
 #endif
 
+extern int force_presence_change(struct platform_device *dev, int state);
+extern int get_host_wake_irq(void);
+extern int wifi_pm_gpio_ctrl(char *name, int level);
+
 struct wifi_platform_data dhd_wlan_control = {0};
 
 #ifdef CUSTOMER_OOB
@@ -23,6 +26,8 @@ uint bcm_wlan_get_oob_irq(void)
 	printf("GPIO(WL_HOST_WAKE) = EXYNOS4_GPX0(7) = %d\n", EXYNOS4_GPX0(7));
 	host_oob_irq = gpio_to_irq(EXYNOS4_GPX0(7));
 	gpio_direction_input(EXYNOS4_GPX0(7));
+#elif defined(CONFIG_ARCH_CPU_SLSI) || defined(CONFIG_ARCH_SUN8IW7)
+    host_oob_irq = get_host_wake_irq();
 #endif
 	printf("host_oob_irq: %d \r\n", host_oob_irq);
 
@@ -33,11 +38,11 @@ uint bcm_wlan_get_oob_irq_flags(void)
 {
 	uint host_oob_irq_flags = 0;
 
-#ifdef CONFIG_MACH_ODROID_4210
+#if defined(GPIO_WLAN_HOST_WAKE) || defined(CONFIG_ARCH_CPU_SLSI) || defined(CONFIG_ARCH_SUN8IW7)
 #ifdef HW_OOB
-	host_oob_irq_flags = IORESOURCE_IRQ | IORESOURCE_IRQ_HIGHLEVEL | IORESOURCE_IRQ_SHAREABLE;
+    host_oob_irq_flags = IORESOURCE_IRQ | IORESOURCE_IRQ_HIGHLEVEL | IORESOURCE_IRQ_SHAREABLE;
 #else
-	host_oob_irq_flags = IORESOURCE_IRQ | IORESOURCE_IRQ_HIGHEDGE | IORESOURCE_IRQ_SHAREABLE;
+    host_oob_irq_flags = IORESOURCE_IRQ | IORESOURCE_IRQ_HIGHEDGE  | IORESOURCE_IRQ_SHAREABLE;
 #endif
 #endif
 	printf("host_oob_irq_flags=%d\n", host_oob_irq_flags);
@@ -50,19 +55,28 @@ int bcm_wlan_set_power(bool on)
 {
 	int err = 0;
 
-	if (on) {
-		printf("======== PULL WL_REG_ON HIGH! ========\n");
-#ifdef CONFIG_MACH_ODROID_4210
-		err = gpio_set_value(EXYNOS4_GPK1(0), 1);
-#endif
-		/* Lets customer power to get stable */
-		mdelay(100);
-	} else {
-		printf("======== PULL WL_REG_ON LOW! ========\n");
-#ifdef CONFIG_MACH_ODROID_4210
-		err = gpio_set_value(EXYNOS4_GPK1(0), 0);
+    if (on) {
+        printf("======== PULL WL_REG_ON HIGH! ========\n");
+#ifdef GPIO_WLAN_EN
+        gpio_set_value(GPIO_WLAN_EN, 1);
+#elif defined(CONFIG_ARCH_CPU_SLSI)
+        wifi_pm_gpio_ctrl("bcmdhd", 1);
+#elif defined(CONFIG_ARCH_SUN8IW7)
+        wifi_pm_gpio_ctrl("wl_reg_on", 1);
+#endif
+        /* Lets customer power to get stable */
+        msleep(50);
+    } else {
+        printf("======== PULL WL_REG_ON LOW! ========\n");
+#ifdef GPIO_WLAN_EN
+        gpio_set_value(GPIO_WLAN_EN, 0);
+#elif defined(CONFIG_ARCH_CPU_SLSI)
+        wifi_pm_gpio_ctrl("bcmdhd", 0);
+#elif defined(CONFIG_ARCH_SUN8IW7)
+        wifi_pm_gpio_ctrl("wl_reg_on", 0);
 #endif
-	}
+        msleep(50);
+    }
 
 	return err;
 }
@@ -71,17 +85,11 @@ int bcm_wlan_set_carddetect(bool present
 {
 	int err = 0;
 
-	if (present) {
-		printf("======== Card detection to detect SDIO card! ========\n");
-#ifdef CONFIG_MACH_ODROID_4210
-		err = sdhci_s3c_force_presence_change(&sdmmc_channel, 1);
-#endif
-	} else {
-		printf("======== Card detection to remove SDIO card! ========\n");
-#ifdef CONFIG_MACH_ODROID_4210
-		err = sdhci_s3c_force_presence_change(&sdmmc_channel, 0);
+#if defined(CONFIG_ARCH_CPU_SLSI) || defined(CONFIG_ARCH_SUN8IW7)
+    force_presence_change(NULL, present);
+#else
+    mmc_force_presence_change_onoff(&sdmmc_channel, present);
 #endif
-	}
 
 	return err;
 }
