--- a/drivers/net/wireless/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/bcmdhd/dhd_linux.c
@@ -4231,24 +4231,9 @@ dhd_preinit_ioctls(dhd_pub_t *dhd)
 	dhd_wl_ioctl_cmd(dhd, WLC_SET_VAR, buf, sizeof(buf), TRUE, 0);
 #endif 
 #endif /* DISABLE_11N */
-#ifdef MIRACAST_APPLICATION
-{
-        int srl = 15;
 
-        ret = dhd_wl_ioctl_cmd(dhd, WLC_SET_SRL, &srl , sizeof(srl), true, 0);
-        if (unlikely(ret)) {
-                WL_ERR(("Set WLC_SET_LRL , error (%d)\n", ret));
-        }
-}
-{
-        int lrl = 15;
 
-        ret = dhd_wl_ioctl_cmd(dhd, WLC_SET_LRL, &lrl , sizeof(lrl), true, 0);
-        if (unlikely(ret)) {
-                WL_ERR(("Set WLC_SET_LRL , error (%d)\n", ret));
-        }
-}
-#endif
+
 	/* query for 'ver' to get version info from firmware */
 	memset(buf, 0, sizeof(buf));
 	ptr = buf;
@@ -6834,4 +6819,4 @@ dhd_wlfc_cleanup2(struct net_device *net
 	dhd_os_wlfc_unblock(&dhd->pub);
 #endif
 }
-#endif
\ No newline at end of file
+#endif
