--- a/drivers/net/wireless/bcmdhd/wl_android.c
+++ b/drivers/net/wireless/bcmdhd/wl_android.c
@@ -37,6 +37,10 @@
 #endif /* CONFIG_WIFI_CONTROL_FUNC */
 
 #ifndef WL_CFG80211
+#error "this is not set"
+#endif
+
+#ifndef WL_CFG80211
 #define htod32(i) i
 #define htod16(i) i
 #define dtoh32(i) i
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -74,3 +74,7 @@ EXTRA_CFLAGS = $(DHDCFLAGS)
 ifeq ($(CONFIG_BCMDHD),m)
 EXTRA_LDFLAGS += --strip-debug
 endif
+
+$(warning "HELPPPPPPPPPPPPPPP")
+$(warning $(CONFIG_CFG80211))
+$(warning $(DHDCFLAGS))
